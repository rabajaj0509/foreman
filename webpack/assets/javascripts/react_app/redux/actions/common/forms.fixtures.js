export const requestData = {
  headers: {},
  item: 'Resource',
  url: '/api/resource',
  values: { a: 1 },
  customErrorAlert: () => {
    return 'Value should not be blank';
  },
};

export const requestDataMsg = Object.assign({}, requestData, {
  message: 'Customized success!',
});
