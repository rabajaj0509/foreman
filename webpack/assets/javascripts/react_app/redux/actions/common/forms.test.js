/* eslint-disable promise/prefer-await-to-then */
import { submitForm } from './forms';
import { mockReset } from '../../../mockRequests';
import { requestData } from './forms.fixtures';
import { testActionSnapshotWithFixtures } from '../../../common/testHelpers';

const fixtures = {
  'should submitForm': () => submitForm(requestData),
};

describe('form actions', () => {
  testActionSnapshotWithFixtures(fixtures);

  beforeEach(() => {
    document.head.innerHTML = `<meta name="csrf-param" content="authenticity_token" />
     <meta name="csrf-token" content="token123" />`;
  });
  afterEach(() => {
    mockReset();
  });

  it('SubmitForm must include an object item/values', () => {
    expect(() => {
      submitForm();
    }).toThrow();
    expect(() => {
      submitForm({ url: 'http://example.com' });
    }).toThrow();
    expect(() => {
      submitForm({ item: 'Resource' });
    }).toThrow();
    expect(() => {
      submitForm({ values: { a: 1 } });
    }).toThrow();
  });
});
