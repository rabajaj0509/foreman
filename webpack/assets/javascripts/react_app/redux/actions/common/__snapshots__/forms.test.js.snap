// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form actions should submitForm 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "actionTypes": undefined,
        "errorToast": [Function],
        "handleError": [Function],
        "handleSuccess": [Function],
        "headers": Object {},
        "key": "RESOURCE_FORM_SUBMITTED",
        "params": Object {
          "a": 1,
        },
        "successToast": [Function],
        "url": "/api/resource",
      },
      "type": "API_POST",
    },
  ],
]
`;
